autopush/router/simple.py

```diff
@@ -137,9 +145,23 @@ def route_notification(self, notification, uaid_data):
             returnValue(RouterResponse(response_body="Delivered"))
         else:
             self.metrics.increment("router.broadcast.miss")
+            if self.udp is not None:
+                yield deferToThread(self._send_udp_wake,
+                                    self.udp)
             returnValue(RouterResponse(202, "Notification Stored"))
 
-    #############################################################
+    def _send_udp_wake(self, udp_info):
+        host = self.conf["udp_server"]
+        response = requests.post(
+            host,
+            data=udp_info["data"],
+            cert=self.conf.get("cert"))
+        if response.status_code < 200 or response.status_code >= 300:
+            raise RouterException("Could not send UDP Wakeup",
+                                  status_code=500,
+                                  response_body="Could not send UDP Wakeup")
```

Comment:
If we successfully stored the message for the user, I'm not sure why we'd return a server error of 500 just because maybe a partner's system to wake is not responding at the moment.